### PR TITLE
Snap to screen pixels instead of px

### LINKED ISSUE #7665: border-width: 0.5px is painted as 1px border
0 5border

### PR DESCRIPTION
**Pull request by mbrubeck**
Fixes #7665. r? @glennw

[Review on Reviewable](https://reviewable.io/reviews/servo/servo/7667)

### CODE REVIEW

**Comment by mbrubeck** [comment]
Note: I haven't tested this on an actual retina-style display, but I simulated it by zooming and by using `--device-pixel-ratio`.

Also, I forgot to add a test, which I'll do shortly.

File: components/gfx/paint_context.rs [hunk 1]
```diff
@@ -1601,7 +1619,7 @@ impl ToAzureSize for Size2D<Au> {
 }
 
 impl ToAzureSize for AzIntSize {
-    fn to_nearest_azure_size(&self) -> Size2D<AzFloat> {
+    fn to_nearest_azure_size(&self, _: f32) -> Size2D<AzFloat> {
```
Comment (glennw) [inline components/gfx/paint_context.rs]: Shouldn't this also apply the scale?

**Comment by mbrubeck** [comment]
`AzIntSize::to_nearest_azure_size` was dead code. Removed in the latest version.

**Comment by mbrubeck** [comment]
@bors-servo retry

#7422

**Pull request merged**